construct/core.py

```diff
@@ -1350,7 +1351,7 @@ def _parse(self, stream, context, path):
         for i,sc in enumerate(self.subcons):
             if sc.flagembedded:
                 subobj = list(sc._parse(stream, context, path).items())
-                obj.update(subobj)
+                ret.update(subobj)
```

Comment:
Argh, yeah it was a bug. I would prefer the variable to be called obj but it fixes it so good job.